weto/src/main/java/fi/uta/cs/weto/actions/forum/ForumActions.java

```diff
@@ -345,9 +393,82 @@ public String action() throws Exception
       tag.setType(TagType.FORUM_TOPIC.getValue());
       tag.insert(conn);
       addActionMessage(getText("forum.message.messageAdded"));
+      // Lets make forum subscription for the teachers, if this is the first topic and the first time subscription for the teacher.
+      if (Tag.selectByTaggedIdAndType(conn, taskId, TagType.FORUM_TOPIC.getValue()).size() == 1) {
+        ArrayList<UserTaskView> teachers = UserTaskView.selectByTaskIdAndClusterType(conn, taskId, ClusterType.TEACHERS.getValue());
+        for (int i = 0; i < teachers.size(); i++) {
+          if (Tag.selectByTaggedIdAndAuthorIdAndType(conn, taskId, teachers.get(i).getUserId(), TagType.FORUM_SUBSCRIPTION.getValue()).isEmpty()) {
+            Tag subscription = new Tag();
+            subscription.setTaggedId(taskId);
+            subscription.setAuthorId(teachers.get(i).getUserId());
+            subscription.setType(TagType.FORUM_SUBSCRIPTION.getValue());
+            subscription.setRank(-1);
+            subscription.insert(conn);
+          }
+        }
```

Comment:
Include code readability
```suggestion
        for (UserTaskView teacher : teachers) {
          if (!Tag.selectByTaggedIdAndAuthorIdAndType(conn, taskId, teacher.getUserId(), TagType.FORUM_SUBSCRIPTION.getValue()).isEmpty()) {
            continue;
          }
          Tag subscription = new Tag();
          subscription.setTaggedId(taskId);
          subscription.setAuthorId(teacher.getUserId());
          subscription.setType(TagType.FORUM_SUBSCRIPTION.getValue());
          subscription.setRank(-1);
          subscription.insert(conn);
        }
```